### ISSUE TITLE
[SF][a11y] ui5-timeline – missing role on list items

### ISSUE THREAD

**Issue by rchrdchn** [SAP/ui5-webcomponents#3794]
### **Bug Description**
Timeline component list items missing `role=list` / `role=option` attribute for screen readers and a11y.

### **Expected Behavior**
Timeline component list items should have `role=list` / `role=option` attribute for screen readers and a11y.

### **Steps to Reproduce**
1. Go to https://sap.github.io/ui5-webcomponents/playground/components/Timeline/
2. Use axe DevTools to check a11y
3. Issue description: `Ensures <li> elements are used semantically`
4. Fix: `List item does not have a <ul>, <ol> parent element without a role, or a role="list"`

### **Isolated Example**
https://sap.github.io/ui5-webcomponents/playground/components/Timeline/

### **Context**
- UI5 Web Components version: @ui5/webcomponents: 0.31.14
- OS/Platform: macOS
- Browser: Chrome
- Affected component: Timeline

### **Log Output / Stack Trace / Screenshots**
![list issue](https://user-images.githubusercontent.com/11576789/131745027-a4d55f37-c422-44f1-972f-04b165606843.png)

![Screen Shot 2021-09-01 at 13 50 46](https://user-images.githubusercontent.com/11576789/131743707-86089f42-88f5-4056-a689-b6412c0f9538.png)

### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

The priority indicates the severity of the issue. To set the appropriate priority consider the following criteria:
- Breaks entire application or system - Low
- Accessibility issue - Medium
- Functional issue - Low
- Visual issue - Low

**Note:** The priority might be re-evaluated by the issue processor.

### **Stakeholder Info** (if applicable)

- Organization: SAP SuccessFactors
- Bussiness impact: a11y screen readers

**Comment by gmkv** [comment]
Hi @SAP/ui5-webcomponents-topic-b 

This is an a11y issue for the Timeline component, where AXE tool reports a missing role.
Can you have a look?

Regards,
Georgi

**Comment by a-z-ivanov** [comment]
Hello @rchrdchn,

Thanks for the detailed explanation. Looking at our queue I found an older issue which is the same.
Thus, I'll close this one and we'll continue working on the older one #3459. There is already a pull request with the solution
https://github.com/SAP/ui5-webcomponents/pull/3650. Also, the pull request contains a discussion on the topic. The solution may change from 'listbox' with 'options' to 'list' with 'items'.

Regards,
Alex

**Comment by ilhan007** [comment]
Hello @rchrdchn with 0.31.20 we introduced the following change in the Timeline component
https://github.com/SAP/ui5-webcomponents/pull/4053/files to fix the a11y issues. 

Actually, our experts suggested removing the "listbox" role to fix the issues, reported by axe core.
- **role="listbox"** removed from **<ul class="ui5-timeline-list"** 
- **role="option"** removed from **<div class="ui5-tli-bubble"**

Did you try out 0.31.20? Is the screenshot above using 0.31.20.

**Comment by ilhan007** [comment]
FYI @unazko @ghristanov could you take a look?

Hello @rchrdchn

Basically, your issue has been closed as duplicate of https://github.com/SAP/ui5-webcomponents/issues/3459
Issue https://github.com/SAP/ui5-webcomponents/issues/3459 has been discussed internally and resolved with this change: https://github.com/SAP/ui5-webcomponents/pull/4053

After that , the change has been downported to 0.31.20.
You can see the generated [Timeline template](https://unpkg.com/browse/@ui5/webcomponents-fiori@0.31.20/dist/generated/templates/TimelineTemplate.lit.js)  in 0.31.20 - notice that the **"ul"** does not have **role="listbox"**.

So, in 0.31.20 we don't set role="listbox". I assume something went wrong with your testing, did you also update the UI5 Web Components For React version to the latest SF version (I think it was 0.131.22)?

**Comment by ilhan007** [comment]
Hello @rchrdchn

these links show the latest stable version that has been released, you can't use them as reference:
https://sap.github.io/ui5-webcomponents/playground/components/Timeline/

Use the "master" version, notice the "master" in the url's path:
https://sap.github.io/ui5-webcomponents/master/playground/components/Timeline/

**Comment by rchrdchn** [comment]
Thanks @ilhan007. It looks like it's been fixed 🙂
